http/client.go

```diff
@@ -336,34 +342,34 @@ func (conn *Connection) execHTTPRequest(req *http.Request, headers map[string]st
 	return status, obj, nil
 }
 
-func closing(c io.Closer) {
+func (conn *Connection) closing(c io.Closer) {
```

Comment:
wouldn't it be neater to pass `conn.log` as a parameter rather than putting `closing` on `conn`? After all, `closing` only needs the `io.Closer`, which isn't necessarily tied to this connection.